python/ctsm/subset_data.py

```diff
@@ -0,0 +1,614 @@
+sys.path.insert(1, _CTSM_PYTHON)
+
+DEFAULTS_FILE = "default_data.cfg"
+
+logger = logging.getLogger(__name__)
+
+
+def get_parser():
+    """
+    Get the parser object for subset_data.py script.
+
+    Returns:
+        parser (ArgumentParser): ArgumentParser which includes all the parser information.
+
+    """
+    parser = ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+
+    parser.print_usage = parser.print_help
+    subparsers = parser.add_subparsers(
+        help="Two possible ways to run this script, either:", dest="run_type"
+    )
+    pt_parser = subparsers.add_parser("point", help="Run script for a single point.")
+    rg_parser = subparsers.add_parser("region", help="Run script for a region.")
+
+    # -- single point parser options
+    pt_parser.add_argument(
+        "--lat",
+        help="Single point latitude. [default: %(default)s]",
+        action="store",
+        dest="plat",
+        required=False,
+        type=plat_type,
+        default=42.5,
+    )
+    pt_parser.add_argument(
+        "--lon",
+        help="Single point longitude. [default: %(default)s]",
+        action="store",
+        dest="plon",
+        required=False,
+        type=plon_type,
+        default=287.8,
+    )
+    pt_parser.add_argument(
+        "--site",
+        help="Site name or tag. [default: %(default)s]",
+        action="store",
+        dest="site_name",
+        required=False,
+        type=str,
+        default="",
+    )
+    pt_parser.add_argument(
+        "--variable-snow-fraction",
+        help="Create surface data with variable snow fraction. If unset, snow is set to uniform "
+             "fraction.",
+        action="store_false",
+        dest="uni_snow",
+        required=False,
+        default=True,
+    )
+    pt_parser.add_argument(
+        "--allow-multiple-pft",
+        help="Create surface data with multiple PFTs. If unset, it assumes the whole grid is 100%% "
+             "single PFT set by --dom-pft.",
+        action="store_false",
+        dest="overwrite_single_pft",
+        required=False,
+        default=True,
+    )
+    pt_parser.add_argument(
+        "--zero-nonveg",
+        help="Set all non-vegetation landunits in the surface data to zero.",
+        action="store_true",
+        dest="zero_nonveg",
+        required=False,
+        default=False,
+    )
+    pt_parser.add_argument(
+        "--allow-saturation-excess",
+        help="Create surface data allowing saturated conditions. If unset saturation excess is set "
+             "to zero.",
+        action="store_true",
+        dest="saturation_excess",
+        required=False,
+        default=False,
+    )
+    pt_parser.add_argument(
+        "--dompft",
+        help="Dominant PFT if we set the grid to 100%% one PFT [default: %(default)s].",
+        action="store",
+        dest="dom_pft",
+        type=int,
+        default=7,
```

Comment:
In the CTSM SE meeting on 2021-01-06, we have decided to keep the default of `dompft` as 7 but changed the defaults of other options as summarized in my comment below: 
https://github.com/ESCOMP/CTSM/pull/1461#issuecomment-1006914348

Reply:
@jkshuman, @adrifoster, @wwieder : can you please confirm you are okay with the default `dompft` as 7? 

Please note that we have changed `--overwrite_single_pft` default to `False` now. 
So the user should explicitly use `--overwrite_single_pft` for it to activate.

Reply:
@negin513 Thanks for checking on this. I think @billsacks has a point that setting this as a default introduces a potential user mismatch.
I think @wwieder question about using overwrite_to_single_pft to get 100% **and** set the dominat pft is worth considering. It removes a potential point of failure.

Reply:
I like @wwieder and @jkshuman 's suggestion here – combining `--overwrite-single-pft` and `--dompft` into a single, optional, integer argument (probably with a default of None).

Reply:
@jkshuman and @wwieder thanks so much for confirming this. 

First, I agree with @wwieder and @jkshuman that using both flags is possibly confusing. 

The `--overwrite-to-single-pft` and`--dompft` are used only in the following lines:
https://github.com/negin513/ctsm/blob/0407486a10ecb693ca1af629609b64e55ce3ec06/python/ctsm/site_and_regional/single_point_case.py#L232-L236

> I think the default `dompft` = 7 is fine.
> 
>  From what I can tell from these notes users have to set both `--overwrite-to-single-pft` and `--dompft` flags to get 100% of a single pft? Is there a case when only one of these flags would be used? If so, should this be described in the help? If not, are both flags really needed?

Yes. I agree. Currently, If the user only specify the flag`--overwrite-to-single-pft`without specifying `--dompft` it will set it to 7.
I am going to remove the flag `--overwrite-to-single-pft` and change `--dompft` default to None. This way if  the use specify `--dompft` to something and change it from None, they will have the behavior of `--overwrite-to-single-pft`.

Basically, changing:
```python
        if self.overwrite_single_pft:
            f_out["PCT_NAT_PFT"][:, :, :] = 0
            if self.dom_pft < 16:
                f_out['PCT_NAT_PFT'][:, :, self.dom_pft] = 100
```
to 
```python
        if self.dom_pft is not None:
            f_out["PCT_NAT_PFT"][:, :, :] = 0
            if self.dom_pft < 16:
                f_out['PCT_NAT_PFT'][:, :, self.dom_pft] = 100
```

Reply:
This is addressed now. (`--overwrite-to-single-pft` and change `--dompft` default to None.)

Reply:
Handled.